js/lib/widget.js

```diff
@@ -254,6 +257,7 @@ export class RemoteFrameBufferView extends DOMWidgetView {
 
         // Scrolling. Need a special throttling that accumulates the deltas.
         // Also, only consume the wheel event when we have focus.
+        // On Firefox, e.buttons is always 0 for wheel events, so we use a cached value for the buttons.
```

Comment:
I read this on MDN under the section [Firefox notes](https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent/buttons#firefox_notes):

> On Mac OS X 10.5, the buttons attribute always returns 0 because there is no platform API for implementing this feature.

Reply:
Ah, that explains it. Though surprising that Firefox does not fake it a way similar to what we do here.

I'd rather just use the cached value always, for consistency/simplicity.

Reply:
I'm concerned it's going to be inaccurate though. If I hold my mouse buttons, move out of the window, release the buttons, move back into the window and scroll the wheel, will the values be correct?

Reply:
Yes, that should work correctly because of mouse capturing. Every mouse down should be followed by a mouse up.

Reply:
https://github.com/vispy/jupyter_rfb/blob/main/js/lib/widget.js#L220-L224